website/docs/reference/resource-properties/constraints.md

```diff
@@ -68,9 +66,81 @@ models:
           - type: ...
 ```
 
+Today, in [dbt Cloud Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can use the new recommended syntax for defining foreign keys. This syntax leverages `ref`. For more information, refer to the section for the [forthcoming v1.9 release](/reference/resource-properties/constraints#whats-better-about-this-syntax).
+
+</File>
+
+</VersionBlock>
+
+<VersionBlock firstVersion="1.9">
+
+## Defining constraints
+
+Constraints may be defined for a single column, or at the model level for one or more columns. As a general rule, we recommend defining single-column constraints directly on those columns.
+
+If you are defining multiple `primary_key` constraints for a single model, those _must_ be defined at the model level. Defining multiple `primary_key` constraints at the column level is not supported. 
+
+The structure of a constraint is:
+- `type` (required): one of `not_null`, `unique`, `primary_key`, `foreign_key`, `check`, `custom`
+- `expression`: Free text input to qualify the constraint. Required for certain constraint types, and optional for others.
+- `name` (optional): Human-friendly name for this constraint. Supported by some data platforms.
+- `columns` (model-level only): List of column names to apply the constraint over
+
+Today, in [dbt Cloud Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can use the new recommended syntax for defining foreign keys. This syntax leverages `ref`.
+
+### What's better about this syntax?
+
+dbt will automatically resolve references to other models so that these references:
+
+- It captures dependencies.
+- Works across environments.
+
+Here are two examples demonstrating how the new syntax is utilized:
+
+<File name='models/schema.yml'>
+
+```yml
+
+models:
+  - name: my_model
+    constraints:
+      - type: foreign_key
+        columns: [id]
+        to: ref('my_model_to') | source('source', 'source_table')
+        to_columns: [id]
+    columns:
+      - name: id
+        data_type: integer
+
+```
+
 </File>
 
+<File name='models/schema.yml'>
+
+```yml
 
+models:
+  - name: my_model
+    columns:
+      - name: id
+        data_type: integer
+        constraints:
+        - type: foreign_key
+          to: ref('my_model_to') | source('source', 'source_table')
+          to_columns: [id]
+
+```
+
+</File>
+
+:::info
+
+The recommended syntax for defining foreign keys will be included in dbt Core v1.9 (open source) when it is released.
```

Comment:
do we think this callout is useful for the user here? wondering if mentioning the 1.9 release in the ["In dbt Cloud Versionless, you can now use.."](https://github.com/dbt-labs/docs.getdbt.com/pull/6105/files#r1768466295) sentence?

Reply:
Hiya @mirnawong1 

Thanks for this suggestion. I'm happy to remove it and mentioning it in the doc you've linked but @jtcohen6 said this in slack: https://dbt-labs.slack.com/archives/C044WU55X7Y/p1726593674040019?thread_ts=1726589431.360079&cid=C044WU55X7Y. I'll wait his update and remove on his ok?

Kind Regards
Natalie

Reply:
Agree with @mirnawong1 - "In dbt Cloud Versionless, you can now use ... This will become available in dbt Core v1.9."